apps/api/src/threads/threads.service.ts

```diff
@@ -1602,53 +1602,89 @@ export class ThreadsService {
         stream,
         initialMessage,
       )) {
-        chunkCount++;
-        if (!ttfbEnded) {
-          ttfbSpan.end();
-          ttfbEnded = true;
-        }
+        const chunkProcessingSpan = Sentry.startInactiveSpan({
+          name: "threads.processStreamChunk",
+          op: "stream.chunk",
+          attributes: {
+            threadId,
+            projectId,
+            chunkNumber: chunkCount + 1,
+            hasToolCall: !!threadMessage.toolCallRequest,
+          },
+        });
 
-        // Update db message on interval
-        const currentTime = Date.now();
-        if (currentTime - lastUpdateTime >= updateIntervalMs) {
-          const isCancelled = await checkCancellationStatus();
+        try {
+          chunkCount++;
+          if (!ttfbEnded) {
+            ttfbSpan.end();
+            ttfbEnded = true;
+          }
 
-          if (isCancelled) {
+          // Update db message on interval
+          const currentTime = Date.now();
+          if (currentTime - lastUpdateTime >= updateIntervalMs) {
+            const isCancelled = await Sentry.startSpan(
+              {
+                name: "threads.checkCancellation",
+                op: "db.query",
+                attributes: { threadId, projectId },
+              },
+              async () => await checkCancellationStatus(),
+            );
+
+            if (isCancelled) {
+              yield {
+                responseMessageDto: {
+                  ...threadMessage,
+                  content: convertContentPartToDto(threadMessage.content),
+                  componentState: threadMessage.componentState ?? {},
+                },
+                generationStage: GenerationStage.CANCELLED,
+                statusMessage: "cancelled",
+                mcpAccessToken,
+              };
+              return;
+            }
+
+            await Sentry.startSpan(
+              {
+                name: "threads.updateMessage",
```

Comment:
Now that we have db spans, I think this span is too little to be useful - just leave it out for now